src/Microsoft.Extensions.Testing.Abstractions/PortablePdbReader.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+
+namespace Microsoft.Extensions.Testing.Abstractions
+{
+    public class PortablePdbReader : IPdbReader
+    {
+        private MetadataReader _reader;
+
+        public PortablePdbReader(Stream pdbStream)
+        {
+            pdbStream.Position = 0;
+            var buffer = new byte[pdbStream.Length];
+            pdbStream.Read(buffer, 0, buffer.Length);
+
+            unsafe
+            {
+                fixed (byte* bufferPtr = buffer)
+                {
+                    _reader = new MetadataReader(bufferPtr, buffer.Length);
```

Comment:
Not good. The memory needs to be pinned for the entire lifetime of MetadataReader. Use GCHandle.Allocate(...).

I need to add helper factories to MetadataReader that create it from Stream, array, etc. to make this easier.